uaclient/actions.py

```diff
@@ -113,8 +114,20 @@ def enable_entitlement_by_name(
     """
     ent_cls = entitlements.entitlement_factory(cfg=cfg, name=name)
     entitlement = ent_cls(
-        cfg, assume_yes=assume_yes, allow_beta=allow_beta, called_name=name
+        cfg,
+        assume_yes=assume_yes,
+        allow_beta=allow_beta,
+        called_name=name,
+        access_only=access_only,
     )
+    if not entitlement.supports_access_only and access_only:
+        msg = messages.ENABLE_ACCESS_ONLY_NOT_SUPPORTED.format(
+            service=entitlement.title
+        )
+        raise exceptions.UserFacingError(
+            msg=msg.msg,
+            msg_code=msg.name,
+        )
```

Comment:
This is just an idea, but could make this check inside the `can_enable` method. I think this would be better since we might not use the `actions` module in the future

Reply:
thats a good idea, I'll move it there!

Reply:
@lucasmoura updated!